net/nimble/host/include/host/ble_store.h

```diff
@@ -131,6 +137,40 @@ union ble_store_value {
     struct ble_store_value_cccd cccd;
 };
 
+struct ble_store_status_event {
+    /**
+     * The type of object that failed to persist; one of the
+     * BLE_STORE_OBJ_TYPE_[...] codes.
+     */
+    int obj_type;
+
+    /**
+     * The type of event being reported; one of the BLE_STORE_EVENT_TYPE_[...]
+     * codes.
+     */
+    int event_code;
+
+    /**
+     * Additional data related to the event; the valid field is inferred from
+     * the obj_type,event_code pair.
+     */
+    union {
+        /**
+         * The record that failed to be written.  Valid for the following event
+         * types:
+         *     o BLE_STORE_EVENT_OVERFLOW
```

Comment:
I didn't think that information is necessary for the reader to understand the code.  The obj_type field isn't part of the union, so is valid for all events.  The union fields, on the other hand, are not always valid; the user needs to know how to determine which one to use.

I am a bit confused about what you are saying here.  Which of these are you saying I should do:

1. Valid for all possible obj_types

2. Valid for the following obj_types:
* BLE_STORE_OBJ_TYPE_OUR_SEC
* BLE_STORE_OBJ_TYPE_PEER_SEC
* BLE_STORE_OBJ_TYPE_CCCD